src/codeu/chat/client/commandline/Chat.java

```diff
@@ -136,8 +148,11 @@ public void invoke(Scanner args) {
     //
     panel.register("u-add", new Panel.Command() {
       @Override
-      public void invoke(Scanner args) {
-        final String name = args.hasNext() ? args.nextLine().trim() : "";
+      public void invoke(List<String> args) {
+        //TODO: possible place of error, not sure how to test username creation
+        // what's in the List being passed in? Is it already tokenized? The scanner code took the entire line?
```

Comment:
To make it clearer what `List<string>` contains, could you instead just pass a `Tokenizer` to all these methods? It seems like it just calls with whatever is passed to `.handleCommand()` up on line 81. I'm not sure if it would work though, either is fine.